main/ble.c

```diff
@@ -179,6 +182,8 @@ static inline void ble_operation_perform(ble_operation_t *operation)
             ESP_GATT_WRITE_TYPE_NO_RSP, ESP_GATT_AUTH_REQ_NONE);
         break;
     case BLE_OPERATION_TYPE_WRITE_CHAR:
+        memcpy(write_buf, operation->value, operation->len);
+        write_buf_len = operation->len;
```

Comment:
Why do we need these static variables? They should be part of the operation and remain there until we decide to dequeue it (after it was successful).